module/Finna/src/Finna/Controller/FileController.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * File Controller
+ *
+ * PHP version 7
+ *
+ * Copyright (C) The National Library of Finland 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org   Main Site
+ */
+namespace Finna\Controller;
+
+use Finna\File\Loader as FileLoader;
+use VuFind\Cache\Manager as CacheManager;
+use VuFind\Record\Loader as RecordLoader;
+use VuFind\Session\Settings as SessionSettings;
+
+/**
+ * File Controller
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://vufind.org   Main Site
+ */
+class FileController extends \Laminas\Mvc\Controller\AbstractActionController
+{
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Record loader
+     *
+     * @var RecordLoader
+     */
+    protected $recordLoader;
+
+    /**
+     * File loader
+     *
+     * @var FileLoader
+     */
+    protected $fileLoader;
+
+    /**
+     * Cache manager
+     *
+     * @var CacheManager
+     */
+    protected $cacheManager;
+
+    /**
+     * Session settings
+     *
+     * @var SessionSettings
+     */
+    protected $sessionSettings;
+
+    /**
+     * Constructor
+     *
+     * @param RecordLoader    $recordLoader Record loader
+     * @param FileLoader      $fileLoader   File loader
+     * @param CacheManager    $cm           Cache manager
+     * @param SessionSettings $ss           Session settings
+     */
+    public function __construct(
+        RecordLoader $recordLoader, FileLoader $fileLoader,
+        CacheManager $cm, SessionSettings $ss
+    ) {
+        $this->recordLoader = $recordLoader;
+        $this->fileLoader = $fileLoader;
+        $this->cacheManager = $cm;
+        $this->sessionSettings = $ss;
+    }
+
+    /**
+     * Download 3D model
+     *
+     * @return \Laminas\Http\Response
+     */
+    public function downloadModelAction()
```

Comment:
Laittaisin tämän RecordControlleriin, kun on kuitenkin tietuekohtainen toiminto. Ja silloin voi ladata tietueen samalla tavalla kuin muissakin kohdissa record controllerissa.

Reply:
Tehdään näin!